bodhi-server/bodhi/server/util.py

```diff
@@ -320,7 +321,11 @@ def sanity_check_repodata(myurl, repo_type):
             # Test comps
             comps = libcomps.Comps()
             try:
-                ret = comps.fromxml_f(repo_info['group'])
+                if repo_info['group'].endswith('xz'):
+                    xml_content = lzma.open(repo_info['group']).read()
+                    ret = comps.fromxml_str(xml_content.decode())
```

Comment:
You might as well just support all the options.

```
                if repo_info['group'].endswith('xz'):
                    xml_content = lzma.open(repo_info['group']).read()
                    ret = comps.fromxml_str(xml_content.decode())
                elif repo_info['group'].endswith('gz'):
                    xml_content = gzip.open(repo_info['group']).read()
                    ret = comps.fromxml_str(xml_content.decode())
                elif repo_info['group'].endswith('zst'):
                    xml_content = zstd.open(repo_info['group']).read()
                    ret = comps.fromxml_str(xml_content.decode())
                elif repo_info['group'].endswith('bz2'):
                    xml_content = bz2.open(repo_info['group']).read()
                    ret = comps.fromxml_str(xml_content.decode())
                else:
                    ret = comps.fromxml_f(repo_info['group'])
```

(I haven't tested that code)